app/components/osc-webhook-triggers/osc-webhook-triggers.html

```diff
@@ -1,38 +1,55 @@
-<ng-form name="secretsForm" class="add-webhook">
+<ng-form name="$ctrl.secretsForm" class="add-webhook">
   <div ng-repeat="trigger in $ctrl.webhookTriggers">
     <div class="add-webhook-row">
-      <ui-select 
-        ng-model="trigger.data.type"
-        ng-disabled="$ctrl.isDeprecated(trigger)"
-        on-select="$ctrl.triggerTypeChange(trigger)"
-        search-enabled="false"
-        title="Select a webhook type"
-        class="select-webhook-type"
-        flex>
-        <ui-select-match placeholder="Webhook type">
-          {{ $select.selected.label }}
-        </ui-select-match>
-        <ui-select-choices repeat="option.label as option in $ctrl.webhookTypesOptions">
-          {{ option.label }}
-        </ui-select-choices>
-      </ui-select>
+      <div class="select-webhook-type">
+        <ui-select 
+          ng-model="trigger.data.type"
+          name="triggerType{{$index}}"
+          ng-disabled="$ctrl.isDeprecated(trigger)"
+          on-select="$ctrl.triggerTypeChange(trigger)"
+          search-enabled="false"
+          title="Select a webhook type"
+          ng-class="{'has-error': $ctrl.secretsForm['triggerType{{$index}}'].$touched && !trigger.data.type && trigger.data[trigger.data.type.toLowerCase()].secretReference.name }"
+          focus-on="triggerTypeFocus{{$index}}"
+          ng-required="!trigger.data.type || trigger.data[trigger.data.type.toLowerCase()].secretReference.name"
```

Comment:
This logic doesn't look right. Type is required if type is not set? Isn't that the same thing as just saying required=true?

Reply:
There was a digest loop when just checking for `trigger.data[trigger.data.type.toLowerCase()].secretReference.name` then the `trigger.data.type` wasnt set. But
> I think it's simpler just to disable the select until the trigger type is selected, which avoids a lot edge cases and unnecessary validation on the type field.

should fix it.